Source/CarthageKit/VersionFile.swift

```diff
@@ -40,7 +40,7 @@ extension CachedFramework: Decodable {
 
 struct VersionFile {
 	let commitish: String
-	let xcodeVersion: String
+	let xcodeVersion: String?
```

Comment:
I think we need to switch this to be the Swift version. Then we can store the Swift version for binary versions of Swift frameworks. (Failure to do so would basically be removing our detection that the Swift version changed.)

There are already methods in CarthageKit to get the Swift version from a compiled framework and from the currently selected Swift version, so I don't think this will be too hard.